core-api/src/test/java/com/optimizely/ab/OptimizelyTest.java

```diff
@@ -2636,6 +2638,277 @@ public void getVariationWithInvalidDatafile() throws Exception {
 
     //======== Notification listeners ========//
 
+    /**
+     * Helper method to return decisionListener
+     **/
+    private DecisionNotificationListener getDecisionListener(final String testType,
+                                                             final String testUserId,
+                                                             final Map<String, ?> testUserAttributes,
+                                                             final Map<String, ?> testDecisionInfo) {
+        return new DecisionNotificationListener() {
+            @Override
+            public void onDecision(@Nonnull String type, @Nonnull String userId, @Nonnull Map<String, ?> attributes, @Nonnull Map<String, ?> decisionInfo) {
+                assertEquals(type, testType);
+                assertEquals(userId, testUserId);
+                assertEquals(attributes, testUserAttributes);
+                for (Map.Entry<String, ?> entry : attributes.entrySet()) {
+                    assertEquals(testUserAttributes.get(entry.getKey()), entry.getValue());
+                }
+                for (Map.Entry<String, ?> entry : decisionInfo.entrySet()) {
+                    assertEquals(testDecisionInfo.get(entry.getKey()), entry.getValue());
+                }
+            }
+        };
+    }
+
+    //======GetEnabledFeatures Notification TESTS======//
+
+    /**
+     * Verify that the {@link Optimizely#getEnabledFeatures(String, Map)}
+     * notification listener of getEnabledFeatures is called with multiple FeatureEnabled
+     */
+    @Test
+    public void getEnabledFeaturesWithListenerMultipleFeatureEnabled() throws Exception {
+        assumeTrue(datafileVersion >= Integer.parseInt(ProjectConfig.Version.V4.toString()));
+
+        Optimizely optimizely = Optimizely.builder(validDatafile, mockEventHandler)
+            .withConfig(validProjectConfig)
+            .build();
+
+        int notificationId = optimizely.notificationCenter.addNotificationListener(NotificationCenter.NotificationType.Decision,
+            new DecisionNotificationListener() {
+                @Override
+                public void onDecision(@Nonnull String type, @Nonnull String userId, @Nonnull Map<String, ?> attributes, @Nonnull Map<String, ?> decisionInfo) {
+                    assertEquals(type, NotificationCenter.DecisionNotificationType.FEATURE.toString());
+                }
+            });
+
+        ArrayList<String> featureFlags = (ArrayList<String>) optimizely.getEnabledFeatures(testUserId,
+            new HashMap<String, String>());
+        assertEquals(2, featureFlags.size());
+
+        verify(mockEventHandler, times(2)).dispatchEvent(any(LogEvent.class));
+
+        assertTrue(optimizely.notificationCenter.removeNotificationListener(notificationId));
```

Comment:
Where is the check for notification listener being called with correct params happening?